lib/AST/Decl.cpp

```diff
@@ -5042,3 +5042,20 @@ getObjCNameForSwiftDecl(const ValueDecl *VD, DeclName PreferredName){
     return {Ctx.getIdentifier(getNameForObjC(VD)), ObjCSelector()};
   }
 }
+
+bool swift::objc_translation::
+isVisibleToObjC(const ValueDecl *VD, Accessibility minRequiredAccess,
+                 bool checkParent) {
+  if (!(VD->isObjC() || VD->getAttrs().hasAttribute<CDeclAttr>()))
+    return false;
+  if (VD->getFormalAccess() >= minRequiredAccess) {
+    return true;
+  } else if (checkParent) {
+    if (auto ctor = dyn_cast<ConstructorDecl>(VD)) {
+      // Check if we're overriding an initializer that is visible to obj-c
```

Comment:
If the overridden decl responds with true to isObjC(), the decl itself will, too. There's no need to check it explicitly. Also please call it the override, not the parent.